src/planscape/plan/tests.py

```diff
@@ -360,49 +360,81 @@ def setUp(self):
 
     def test_missing_user(self):
         response = self.client.post(
-            reverse('plan:create_project'), {'max_cost': 100},
+            reverse('plan:create_project'), {},
             content_type='application/json')
         self.assertEqual(response.status_code, 400)
 
     def test_missing_plan(self):
         self.client.force_login(self.user)
         response = self.client.post(
-            reverse('plan:create_project'), {'max_cost': 100},
+            reverse('plan:create_project'), {},
             content_type='application/json')
         self.assertEqual(response.status_code, 400)
 
     def test_null_user_cannot_create_project_with_user(self):
         PLANSCAPE_GUEST_CAN_SAVE = False
         response = self.client.post(
             reverse('plan:create_project'), {
-                'plan_id': self.plan_with_user.pk, 'max_cost': 100},
+                'plan_id': self.plan_with_user.pk},
             content_type='application/json')
         self.assertEqual(response.status_code, 400)
 
     def test_user_cannot_create_project_with_null_user(self):
         self.client.force_login(self.user)
         response = self.client.post(
             reverse('plan:create_project'), {
-                'plan_id': self.plan_no_user.pk, 'max_cost': 100},
+                'plan_id': self.plan_no_user.pk},
             content_type='application/json')
         self.assertEqual(response.status_code, 400)
 
     def test_no_user(self):
         PLANSCAPE_GUEST_CAN_SAVE = False
         response = self.client.post(
             reverse('plan:create_project'), {
-                'plan_id': self.plan_no_user.pk, 'max_cost': 100},
+                'plan_id': self.plan_no_user.pk},
             content_type='application/json')
         self.assertEqual(response.status_code, 200)
 
     def test_good_with_user(self):
         self.client.force_login(self.user)
         response = self.client.post(
             reverse('plan:create_project'), {
-                'plan_id': self.plan_with_user.pk, 'max_cost': 100},
+                'plan_id': self.plan_with_user.pk},
             content_type='application/json')
         self.assertEqual(response.status_code, 200)
 
+    def test_treatment_ratio_bad_format(self):
+        self.client.force_login(self.user)
+        response = self.client.post(
+            reverse('plan:create_project'), {
+                'plan_id': self.plan_with_user.pk, 'max_treatment_ratio': 4},
+            content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+
+    def test_treatment_ratio_bad_format_negative(self):
+        self.client.force_login(self.user)
+        response = self.client.post(
+            reverse('plan:create_project'), {
+                'plan_id': self.plan_with_user.pk, 'max_treatment_ratio': -1},
+            content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+
+    def test_max_slope_bad_format(self):
```

Comment:
I just realized max_slope, in theory, could be greater than 1.0 (contrary to what I wrote in the doc delineating params).

Reading into it, max grade for trucks tends to be 0.12 to 0.15, though, so leaving it as 1.0 is fine.

Just commenting because I don't know what max value would draw the line between plausible and nonsensical.

Reply:
My vote is to take out validation of large positive values in the backend, and raise an error or warning in the frontend when the user sets above 1.0.  Plus, we may get feedback to use "percentage grade" rather than "y/x".

But negative slopes are bad.  Keep checking for those.

Reply:
sg, i'll make that change @thequeenofspades FYI